### PR TITLE
feat: nixbuild.net trait

### PR DESCRIPTION
**Pull request by renebarbosafl**
## Summary by CodeRabbit

- **New Features**
	- Introduced a new integration for connecting to Nixbuild.net, enhancing the deployment capabilities with Nix and SSH configurations.

### CODE REVIEW

**Comment by renebarbosafl** [comment]
atlantis apply

**Comment by renebarbosafl** [comment]
Tests: [tests.zip](https://github.com/input-output-hk/sc-dev-platform/files/13983107/tests.zip)

File: infra/modules/kubevela-addons/definitions/nixbuildnet.cue [hunk 1]
```diff
@@ -0,0 +1,94 @@
+parameter: {
+    secretName: string
```
Comment (shlevy) [inline infra/modules/kubevela-addons/definitions/nixbuildnet.cue]: Ideally this wouldn't be needed, the secret would just be available everywhere (perhaps using the external-secrets thing)

Comment (renebarbosafl) [reply infra/modules/kubevela-addons/definitions/nixbuildnet.cue]: So can we assume the secret name will always be "nbn"? Do you suggest something else?
I can leave this parameter as optional with something as default value.

Comment (renebarbosafl) [reply infra/modules/kubevela-addons/definitions/nixbuildnet.cue]: @shlevy Implemented on https://github.com/input-output-hk/sc-dev-platform/pull/126/commits/ab43c58660c07dbbbaac20b1cbe60851a213f457

**Comment by renebarbosafl** [comment]
atlantis apply

File: infra/modules/kubevela-addons/definitions/nixbuildnet.cue (hunk 1 above)
Comment (shlevy) [reply infra/modules/kubevela-addons/definitions/nixbuildnet.cue]: We can't assume `nbn` unless we also make sure the secret is always available.

Comment (renebarbosafl) [reply infra/modules/kubevela-addons/definitions/nixbuildnet.cue]: hmmm thinking...

If we leave "_nbn_" as the default secret name and it's not available in the namespace, the container will likely fail with `CreateContainerConfigError` until the necessary secret becomes available. The same behavior will happen if the Application is setting parameter.secretName as "_somethingelse_" and the secret "_somethingelse_" is not available.

As we don't have a default mechanism to create/distribute secrets yet I am inclined to keep that behavior and add some notes in the documentation about the secret dependency. Once we have a default mechanism for the secrets management, we can update the trait to perform it automatically.

What do you think about ?

Comment (shlevy) [reply infra/modules/kubevela-addons/definitions/nixbuildnet.cue]: Sounds good

**Pull request closed without merging**